cmd/juju/common/flags.go

```diff
@@ -19,8 +19,16 @@ import (
 // ConfigFlag records k=v attributes from command arguments
 // and/or specified files containing key values.
 type ConfigFlag struct {
-	files []string
-	attrs map[string]interface{}
+	files        []string
+	attrs        map[string]interface{}
+	preserveType bool
+}
+
+// SetPreserveType sets whether name values should be
```

Comment:
Based on how preserveType is used, I don't understand this at all. Is it meant to be preserving the string type? Or inferring the type based on the value's contents (i.e. decoding as YAML)?

Perhaps be a bit more literal, and make this

```go
// SetDecodeYAMLValues controls whether or not config values
// are decoded as YAML into their default types, or to store the
// original string values. The default is to decode as YAML.
func (*ConfigFlag) SetDecodeYAMLValues(bool)
```

Reply:
Changed to "SetPreserveStringValue" and expanded the comment to hopefully clarify.